s3/replication/common/src/s3replicationcommon/s3_get_bucket_replication.py

```diff
@@ -0,0 +1,213 @@
+
+    @staticmethod
+    def prepare_matched_rule_object(rule):
+        """Initialise the attributes from matched rules"""
+        policy_obj = MatchedRuleAttributes()
+
+        if 'DeleteMarkerReplication' in rule:
+            if 'Status' in rule['DeleteMarkerReplication']:
+                policy_obj._delete_marker_replication_status = \
+                        rule['DeleteMarkerReplication']['Status']
+        if 'Destination' in rule:
+            if 'Bucket' in rule['Destination']:
+                policy_obj._dest_bucket = rule['Destination']['Bucket'].split(
+                    ':')[-1]
+            if 'EncryptionConfiguration' in rule['Destination']:
+                policy_obj._encryption_replication_key_id = \
+                        rule['Destination'][
+                            'EncryptionConfiguration']['ReplicaKmsKeyID']
+            if 'Account' in rule['Destination']:
+                policy_obj._account = rule['Destination']['Account']
+            if 'ReplicationTime' in rule['Destination']:
+                policy_obj._replicationtime_status = \
+                        rule['Destination']['ReplicationTime']['Status']
+        if 'Status' in rule:
+            policy_obj._status = rule['Status']
+        if 'Filter' in rule.keys():
+            if 'Prefix' in rule['Filter'].keys():
+                policy_obj._prefix = rule['Filter']['Prefix']
+            if 'Tag' in rule['Filter'].keys():
+                policy_obj._tag = rule['Filter']['Tag']
+        if 'ID' in rule.keys():
+            policy_obj._id = rule['ID']
+        if 'Priority' in rule:
+            policy_obj._priority = rule['Priority']
+        return policy_obj
+
+    def get_replication_rule(self, obj_name):
+        """Returns replication replication rule for given bucket."""
+        self._dest_bucket = None
+        try:
+            for key, value in (
+                    self._response_dict['ReplicationConfiguration']).items():
+                if key == 'Rule':
+                    # Check if whether 'value' instance is list of rules
+                    if isinstance(value, list):
+                        # Iterate through different rules
+                        for rule in value:
+                            # Check if object name marches any rule prefix
+                            if rule['Filter']['Prefix'] in obj_name:
+                                return S3AsyncGetBucketReplication.prepare_matched_rule_object(
+                                    rule)
+                    # If only one rule is present
+                    else:
+                        if value['Filter']['Prefix'] in obj_name:
+                            return S3AsyncGetBucketReplication.prepare_matched_rule_object(
+                                value)
+
+        except Exception as e:
+            self._logger.error(
+                "Failed to get rule! Exception type : {}".format(e))
+
+    # yields data chunk for given size
```

Comment:
move this comment to after first line of method as docstring

Reply:
done